_posts/2018-03-09-code_sample_vanity.md

```diff
@@ -48,6 +49,104 @@ namespace teststrat2
 }
 ```
 
+## Multi-threaded generation and input validation
+
+So you want to use all the CPU available on your computer to generate a fancy vanity address? 
+Here is how you can do that multi-threaded, and there is additional input validation since O 
+and I are not allows in address due to ambiguity. 
+
+Additionally this example does generation based upon input variables, so it's more easy to use 
+without recompiling and you can even share it with your friends. Start first with two characters, and you 
+will quickly learn that it is much harder to generate 3 characters, and even harder with more.
+
+Sometimes the threads find a match almost at the same time, so multiple private key and addresses might 
+appear in the console output.
+
+Example on performance with threads: Generating address with prefix "SASA" with a single thread took
+22 seconds on high-end CPU. With 8 threads, it took 1-2 seconds. Results may vary on different computers,
+and there is an element of (random) luck involved.
+
+
+```cs
+using System;
+using NBitcoin;
+
+namespace teststrat2
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            if (args.Length == 0)
+            {
+                Console.WriteLine("You must supply the address prefix. You can also specify number of threads, as the second parameter. E.g. \"dotnet teststart2.dll \"HI\" 10\".");
+                return;
+            }
+
+            var prefix = args[0];
+
+            var prefixUpper = prefix.ToUpper();
+
+            if (prefixUpper.Contains("O") || prefixUpper.Contains("I") || prefixUpper.Contains("0"))
+            {
+                Console.WriteLine("The characters O, I and 0 (zero) is not allowed in addresses.");
```

Comment:
Only 0 is not allowed. Additionally, A-M and l-z are not allowed in the second character position (after "S"). Consider a comment to this effect.

Reply:
actually, O is not available in the 2nd letter slot specifically, because both lowercase and uppercase o are not available. "i" is allowed however